pkg/pushsync/metrics.go

```diff
@@ -102,6 +103,12 @@ func newMetrics() metrics {
 			Name:      "invalid_stamps",
 			Help:      "No of invalid stamp errors.",
 		}),
+		StampValidationTime: *prometheus.NewGaugeVec(prometheus.GaugeOpts{
```

Comment:
why not histogram?

Reply:
validation duration should be more or less a static value but I don't mind changing it to a histogram to plot fluctuations if they do ever happen.